### DISCUSSION TITLE
Having problems with xxhash_asm.go and xxhash_other.go

### DISCUSSION THREAD

**Discussion by davidbudiman-grabtaxi** [klauspost/compress#698] [Q&A] [answered]
Hi there!

I am fairly new to Golang. 

So, there is this code in our repository that uses your library and it's giving out this error.
<img width="1065" alt="image" src="https://user-images.githubusercontent.com/94107506/203936676-c2576355-731b-49cc-a076-7f0d0a029cc1.png">

Now, I only know that this code is made to support assembly and Go, but I don't know how to resolve this. Please help

**Comment by klauspost** [comment]
Could you post output from `go version` and `go env`? @david-budiman-grab

**Reply by davidbudiman-grabtaxi** [reply]
go version go1.19.2 darwin/arm64

`GO111MODULE="auto"
GOARCH="arm64"
GOBIN=""
GOCACHE="/Users/david.budiman/Library/Caches/go-build"
GOENV="/Users/david.budiman/Library/Application Support/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="arm64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="/Users/david.budiman/gopath/pkg/mod"
GONOPROXY="gitlab.myteksi.net"
GONOSUMDB="gitlab.myteksi.net"
GOOS="darwin"
GOPATH="/Users/david.budiman/gopath"
GOPRIVATE="gitlab.myteksi.net"
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/opt/homebrew/Cellar/go/1.19.2/libexec"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/opt/homebrew/Cellar/go/1.19.2/libexec/pkg/tool/darwin_arm64"
GOVCS=""
GOVERSION="go1.19.2"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD=""
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/pm/mq6v8pzn5ns72gjq_27zh7j80000gq/T/go-build1754502929=/tmp/go-build -gno-record-gcc-switches -fno-common"
`

**Reply by davidbudiman-grabtaxi** [reply]
```
GO111MODULE="auto"
GOARCH="arm64"
GOBIN=""
GOCACHE="/Users/david.budiman/Library/Caches/go-build"
GOENV="/Users/david.budiman/Library/Application Support/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="arm64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="/Users/david.budiman/gopath/pkg/mod"
GONOPROXY="gitlab.myteksi.net"
GONOSUMDB="gitlab.myteksi.net"
GOOS="darwin"
GOPATH="/Users/david.budiman/gopath"
GOPRIVATE="gitlab.myteksi.net"
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/opt/homebrew/Cellar/go/1.19.2/libexec"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/opt/homebrew/Cellar/go/1.19.2/libexec/pkg/tool/darwin_arm64"
GOVCS=""
GOVERSION="go1.19.2"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD=""
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/pm/mq6v8pzn5ns72gjq_27zh7j80000gq/T/go-build1754502929=/tmp/go-build -gno-record-gcc-switches -fno-common"

```

**Comment by klauspost** [comment]
Hmmmm.. Which version of the package are you using? Are you using any build tags?

**Comment by davidbudiman-grabtaxi** [comment, marked as answer]
I found the answer. Maybe because of bad connection, I don't know. But your lib wos not downloaded correctly and there were missing files. After I do a re-clone of your lib, it's not showing the error again.

**Reply by klauspost** [reply]
Great! Quite a mystery to me :D

**Reply by davidbudiman-grabtaxi** [reply]
Some mysteries are not to be answered. Thanks for your fast responses 😄
